components/ui/pathogen-map/pathogen-map-popup.tsx

```diff
@@ -0,0 +1,102 @@
+import { MapRef, Popup, useMap } from "react-map-gl";
+
+import { PathogenDataPointPropertiesBase } from "./pathogen-map";
+
+export type PopupContentGenerator<
+  TPathogenDataPointProperties extends PathogenDataPointPropertiesBase
+> = (input: TPathogenDataPointProperties) => React.ReactNode;
+
+type VisiblePopupInfo<
+  TPathogenDataPointProperties extends PathogenDataPointPropertiesBase
+> = { visible: true; properties: TPathogenDataPointProperties };
+type HiddenPopupInfo<
+  TPathogenDataPointProperties extends PathogenDataPointPropertiesBase
+> = { visible: false; properties: null };
+
+export type PopupInfo<
+  TPathogenDataPointProperties extends PathogenDataPointPropertiesBase
+> =
+  | VisiblePopupInfo<TPathogenDataPointProperties>
+  | HiddenPopupInfo<TPathogenDataPointProperties>;
+
+interface PathogenMapPopupProps<
+  TPathogenDataPointProperties extends PathogenDataPointPropertiesBase
+> {
+  mapId: string;
+  popUpInfo: PopupInfo<TPathogenDataPointProperties>;
+  generatePopupContent: PopupContentGenerator<TPathogenDataPointProperties>;
+}
+
+export function PathogenMapPopup<
+  TPathogenDataPointProperties extends PathogenDataPointPropertiesBase
+>({
+  mapId,
+  popUpInfo,
+  generatePopupContent,
+}: PathogenMapPopupProps<TPathogenDataPointProperties>) {
+  const allMaps = useMap();
+  const map = allMaps[mapId];
+
+  const getMapboxLatitudeOffset = (map: MapRef) => {
+    // Map seems to zoom in in powers of 2, so reducing offset by powers of 2 keeps the modal apprximately
+    // in the same center everytime
+    // offset needs to reduce exponentially with zoom -- higher zoom x smaller offset
+    let mapZoom = map.getZoom();
+
+    return 80 / Math.pow(2, mapZoom);
+  };
+
+  if (popUpInfo.visible === false) {
+    return null;
+  }
+
+  // Not a typo, there is some kind of underlying problem here worth investigating that
```

Comment:
Also an issue I have seen before (We had lat and long switched in the other map). I wonder if we are just calculating them incorrectly or smthn in the ETL